# Empleabilidad Ecuador



In [1]:
# import libraries needed for this job
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

## 1. Lectura y Carga del dataframe

Con la ayuda de técnicas de webscrapping se generó un dataframe que contiene información de varias plataformas de ofertas de trabajo, que incluye información del Cargo, Provincia, Empresa, Descripción, Plataforma y URL.

In [2]:
# read csv file
jobs_df = pd.read_csv("jobs_csv_limpio.csv")
jobs_df.head()

Cargo  Provincia              Empresa  \
0  ASISTENTE DE OFICINA QUITO  Pichincha      Banco Solidario   
1         CERVECERIA NACIONAL       Loja  CERVECERIA NACIONAL   
2                        HLGP     Guayas                 HLGP   
3      Se busca Recepcionista     Guayas         Confidencial   
4  Recepcionista y facturista     Manabí         Confidencial   

                                         Descripción Plataforma  \
0  Nos encontramos en la busqueda de un ASISTENTE...     Indeed   
1  Asegurar disponibilidad del portafolio de prod...     Indeed   
2          SEGURIDAD INDUSTRIAL Y SALUD OCUPACIONAL.     Indeed   
3  Atención a clientes, manejo de archivo, manejo...     Indeed   
4          Comercial / Ventas / Atención al Cliente.     Indeed   

                                                 URL  
0  https://ec.indeed.com/rc/clk?jk=d8afebe5bca3ac...  
1  https://ec.indeed.com/rc/clk?jk=983f478c08551c...  
2  https://ec.indeed.com/rc/clk?jk=4f9815af7f048f...  
3  https://ec.indeed.com/rc/clk?jk=4afc41374880c2...  
4  https://ec.indeed.com/rc/clk?jk=dea4b460c6ce1c...

In [3]:
# Check dataframe structure
jobs_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5574 entries, 0 to 5573
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Cargo        5574 non-null   object
 1   Provincia    5574 non-null   object
 2   Empresa      5574 non-null   object
 3   Descripción  5574 non-null   object
 4   Plataforma   5574 non-null   object
 5   URL          5574 non-null   object
dtypes: object(6)
memory usage: 261.4+ KB


* Este daframe tiene 5574 filas y 6 columnas.
* Todas las variables son del tipo `object`.

In [4]:
# Missing Values?
jobs_df.isna().sum()

Cargo          0
Provincia      0
Empresa        0
Descripción    0
Plataforma     0
URL            0
dtype: int64

No hay missing values.

In [5]:
# How many jobs are there per `Provincia`?
jobs_prov = jobs_df.iloc[:, 0:2]
jobs_prov.groupby('Provincia').count().sort_values('Cargo', ascending=False)

Cargo
Provincia                            
Guayas                           1772
Pichincha                        1515
Ecuador                          1444
Azuay                             230
Tungurahua                        147
Loja                              118
Manabí                             69
Los Ríos                           49
Santo Domingo de los Tsáchilas     42
Cotopaxi                           30
El Oro                             28
Imbabura                           26
Chimborazo                         22
Esmeraldas                         18
Sucumbíos                          11
Santa Elena                        11
Orellana                           10
Cañar                               9
Carchi                              7
Morona Santiago                     7
Pastaza                             3
Bolívar                             2
Galápagos                           2
Napo                                1
Zamora Chinchipe                    1

La Provincia que más ofertas de trabajo tiene es Guayas, seguida por Pichincha. En tercer lugar está Ecuador y esto se refiere a que existen ofertas de trabajo del extranjero y por lo tanto no están referenciadas a una provincia sino a to el país. Las tres provincias más importantes del Ecuador son Pichincha, Guayas y Azuay, lo cual se refleja en tabla anteriormente presentada ya que la mayor cantidad de ofertas de trabajo están concetradas en estas provincias.

## 2. Análisis de Carreras Digitales Técnicas

Esta primera parte del análisis se enfoca en las carreras digitales técnicas, para lo cual se utilizará un vector que previamente fue generado y que contiene este tipo de carreras y que serán consideradas para realizar una búsqueda y agrupación en la columna `Cargo` del dataframe jobs_df. 

In [6]:
# Value count of 'Cargo' column
job = jobs_df['Cargo'].value_counts().reset_index()
job

index  Cargo
0           Ayudante de Obra Civil - Guayaquil    102
1          ASISTENTE DE COMUNICACION -EVENTUAL    102
2                    Ayudante de bodega - Loja    102
3     Ecuador country manager | Ecuador | 2020    101
4            Maestro de Obra Civil - Guayaquil    101
...                                        ...    ...
2909           Ejecutivo/a comercial - Seguros      1
2910           Ayudante de Reclamos - Eventual      1
2911           ASISTENTE DE MARKETING Y VENTAS      1
2912              COORDINADOR DE IMPORTACIONES      1
2913                     Experience Supervisor      1

[2914 rows x 2 columns]

Se aprecia que los cargos no están expresados de manera consistente sino que se incluye información adicional como la ciudad o provincia, el género, en otros.
Por lo anterior, y con la ayuda del vector `possible_tech_jobs` se realizará una búsqueda en la columna `Cargo` de tal manera de extraer solamente las carreras o profesiones requeridas como tal.

In [7]:
# Posible tech digital jobs
possible_tech_jobs = ['Administrador Sistemas', 
                      'Analista de Satos', 
                      'Arquitecto de Software', 
                      'Científico de Datos',
                      'Community Manager', 
                      'Data Scientist', 
                      'Consultor TIC', 
                      'Desarrollador Full Stack', 
                      'Desarrollador Backend',
                      'Desarrollador de Software', 
                      'Front end', 
                      'Desarrollador Móvil', 
                      'Desarrollador Videojuegos',
                      'Web', 
                      'Product Manager', 
                      'Project Manager', 
                      'Diseñador Gráfico', 
                      'Administrador de Bases de Datos',
                      'E-Commerce', 
                      'Especialista SEO', 
                      'Especialista SEM', 
                      'Especialista UI', 
                      'Especialista UX', 
                      'Marketing Digital', 
                      'Ciberseguridad', 
                      'Ingeniero de Datos', 
                      'Data Engineer',
                      'QA',
                      'UX Designer − Diseñador Web', 
                      'Ing. de Sistemas', 
                      'SCRUM MASTER', 
                      'Web Developer', 
                      'Telemercadeo', 
                      'warehouse', 
                      '.NET Developer', 
                      'Diseñador Web', 
                      'Desarrollador App Mobile', 
                      'Remote UX/UI Graphic Designer', 
                      'Experto tecnologico', 
                      'Programador Apis, Sql', 
                      'PuntoNet', 
                      'Programador Senior',  
                      'Ingeniero/a Estadistica', 
                      'Programador PHP', 
                      'Diseñador Web WordPress', 
                      'Data Science Consultant', 
                      'Business Consultant', 
                      'Analista Digital', 
                      'Programador ODOO', 
                      'Desarrollador Software JAVA', 
                      'Android Nativo', 
                      'React js', 
                      'Programador Senior Aplicaciones Móviles', 
                      'Python SSR/SR', 
                      'Java SSR/SR', 
                      'Desarrollador Senior React JS Laravel', 
                      'DESARROLLADOR JAVASCRIPT', 
                      'Programador Javascript', 
                      'ANALISTA Q/A SOFTWARE', 
                      'Programador de software', 
                      'Ingeniero de software', 
                      'Desarrollador de Software.net', 
                      'Business Analyst', 
                      'Business Intelligence']

In [8]:
# The clean up process starts extracting the possible rules 
job_group_tech = job['index'].str.extractall('({0})'.format('|'.join(possible_tech_jobs)))

# Create a column name
job_group_tech.columns = ['new_job']

# Count the number of the possible rules and rename the dataframe
job_group_tech = pd.DataFrame(job_group_tech['new_job'].value_counts()).reset_index()
job_group_tech.rename(columns={'index': 'job', 'new_job': 'count'}, inplace=True)

# Include the percentage
job_group_tech['percentage'] = job_group_tech['count'] / np.sum(job_group_tech['count'])
job_group_tech

job  count  percentage
0                                     Web     18    0.187500
1                                      QA     11    0.114583
2                       Community Manager      9    0.093750
3                       Marketing Digital      7    0.072917
4                           Diseñador Web      4    0.041667
5                         Programador PHP      3    0.031250
6                       Diseñador Gráfico      3    0.031250
7                      Programador Senior      3    0.031250
8                         Project Manager      3    0.031250
9                                React js      2    0.020833
10  Desarrollador Senior React JS Laravel      2    0.020833
11                         .NET Developer      2    0.020833
12                  Business Intelligence      2    0.020833
13                       Business Analyst      2    0.020833
14                             E-Commerce      2    0.020833
15               Desarrollador App Mobile      1    0.010417
16                       Analista Digital      1    0.010417
17                        Product Manager      1    0.010417
18                           Telemercadeo      1    0.010417
19              Desarrollador de Software      1    0.010417
20                               PuntoNet      1    0.010417
21                              Front end      1    0.010417
22               DESARROLLADOR JAVASCRIPT      1    0.010417
23                          Python SSR/SR      1    0.010417
24                  Ingeniero de software      1    0.010417
25                         Android Nativo      1    0.010417
26                           SCRUM MASTER      1    0.010417
27                 Programador Javascript      1    0.010417
28            Desarrollador Software JAVA      1    0.010417
29                            Java SSR/SR      1    0.010417
30                  ANALISTA Q/A SOFTWARE      1    0.010417
31                  Programador Apis, Sql      1    0.010417
32                          Data Engineer      1    0.010417
33                    Experto tecnologico      1    0.010417
34                              warehouse      1    0.010417
35          Remote UX/UI Graphic Designer      1    0.010417
36                       Ing. de Sistemas      1    0.010417
37                Programador de software      1    0.010417

El término `Web` es el que más aparece en esta búsqueda y está relacionado con: Desarrolador Web, Web Master, Programador Web, Web Developer, entre otros. 

## 3. Análisis de Carreras No Digitales Técnicas

La segunda parte del análisis se enfoca en las carreras no digitales técnicas, para lo cual se utilizará otro vector que contiene las carreras no digitales técnicas que serán consideradas para realizar una búsqueda y agrupación en la columna `Cargo` del dataframe jobs_df. 

In [9]:
non_tech_jobs = ['Abogada',
                 'Abogado',
                 'Administrador',
                 'Agente',
                 'Albañil',
                 'Almacenista',
                 'Analista',
                 'Arquitecta',
                 'Arquitecto',
                 'Asesor ',
                 'Asesoria Academica',
                 'Asesoría jurídica',
                 'Asistente',
                 'Auditor',
                 'Auxiliar', 
                 'Ayudante',
                 'Bachiller',
                 'Biólogo',
                 'Bioquímico',
                 'Bodeguero',
                 'Cajera',
                 'Cajero',
                 'Camarera',
                 'Camarero',
                 'Capacitador',
                 'Carpintero',
                 'Chef',
                 'Chofer',
                 'Cobrador',
                 'Cocinero',
                 'Cocturera ',
                 'Comunicador',
                 'Comunicadora ',
                 'Conductor',
                 'Conserje',
                 'Conserje ',
                 'Consultor',
                 'Contador',
                 'Controlador',
                 'Coordinador',
                 'Cosmetóloga',
                 'Costurera',
                 'Country Manager',
                 'CPA ',
                 'Dermatólogo/a Centro',
                 'Despachador',
                 'Dibujante',
                 'Digitador',
                 'Director',
                 'Directora ',
                 'distribuidor',
                 'Docente',
                 'Editor',
                 'Ejecutiva',
                 'Ejecutivo',
                 'Electricista',
                 'Electromecánico',
                 'Empacadores',
                 'Empleada doméstica',
                 'Emprendimiento',
                 'Encuestador',
                 'Enderezador',
                 'Enfermera',
                 'Enfermero',
                 'Especialista',
                 'Estibador',
                 'Gerente',
                 'Gestor',
                 'Guardia de Seguridad',
                 'Guia Turistico',
                 'Impulsadora ',
                 'Ing.',
                 'Ingeniera',
                 'Ingeniero',
                 'Inspector',
                 'Instalador dispositivos GPS',
                 'Instructor',
                 'Investigador',
                 'Jefe',
                 'Key account manager',
                 'Laboratorista',
                 'Laminador',
                 'Licenciado',
                 'Lider',
                 'Liquidador',
                 'Major Accounts Manager',
                 'Manicurista',
                 'Markenting bilingue',
                 'Mecánico',
                 'Médico',
                 'Mensajero',
                 'Mercaderista',
                 'Mesero',
                 'Nutricionista',
                 'Obrera',
                 'Obrero',
                 'Odontólogo',
                 'Oficial',
                 'Operador',
                 'Operario',
                 'Optómetra',
                 'Ortodoncista',
                 'Paletizador',
                 'Panadero',
                 'Pasante',
                 'Perchero',
                 'Periodistas',
                 'Periodoncista',
                 'Pizzero',
                 'Planificador',
                 'Pollero',
                 'Potabilizador',
                 'Psicóloga',
                 'Psicólogo',
                 'Productor',
                 'Profesor',
                 'Promotor',
                 'Recepcionista',
                 'Recursos Humanos',
                 'Redactores',
                 'Relacionista',
                 'Repartidor',
                 'Reportero ',
                 'Representante',
                 'Residente de obra',
                 'Secretaria',
                 'Sexólogo/a',
                 'Social media manager',
                 'Soldador',
                 'ubcontador/a',
                 'Subgerente ',
                 'Supervisor',
                 'Técnico',
                 'Tecnólogo',
                 'Telemercaderista',
                 'Teleoperador',
                 'Teleoperadora',
                 'Teletrabajo',
                 'Teletrabajo ',
                 'Terapista',
                 'Traductor ',
                 'Trainee ',
                 'Vendedor',
                 'Veterinario ',
                 'Vigilante ',
                 'Visitador Médico']

In [10]:
# The clean up process starts extracting the possible rules 
job_group_nontech = job['index'].str.extractall('({0})'.format('|'.join(non_tech_jobs)))

# Create a column name
job_group_nontech.columns = ['new_job']

# Count the number of the possible rules and rename the dataframe
job_group_nontech = pd.DataFrame(job_group_nontech['new_job'].value_counts()).reset_index()
job_group_nontech.rename(columns={'index': 'job', 'new_job': 'count'}, inplace=True)

# Include the percentage
job_group_nontech['percentage'] = job_group_nontech['count'] / np.sum(job_group_nontech['count'])
job_group_nontech.head(20)

job  count  percentage
0       Asistente    147    0.088929
1        Vendedor    100    0.060496
2       Ejecutivo     77    0.046582
3         Asesor      75    0.045372
4         Técnico     73    0.044162
5        Auxiliar     70    0.042347
6            Jefe     67    0.040532
7     Coordinador     66    0.039927
8        Analista     65    0.039322
9      Supervisor     58    0.035088
10           Inge     54    0.032668
11        Pasante     49    0.029643
12       Operador     32    0.019359
13   Mercaderista     29    0.017544
14  Administrador     26    0.015729
15         Médico     25    0.015124
16       Contador     24    0.014519
17        Docente     22    0.013309
18        Gerente     21    0.012704
19       Ayudante     21    0.012704

Se debe tomar en cuenta que para este prototipo se han agrupado los diferentes cargos, como por ejemplo, el cargo de asistente incluye asistente contable, comercial, administrativo, calidad, etc. Esta misma consideración aplica a los diferentes cargos.

## 4. Análisis de Habilidades

La cuarta parte del análisis se refiere a las habilidades requeridas en los diferentes cargos técnicos digitales, para lo cual y al igual que en los anteriores casos vamos a utilizar una lista previamente definida pero la búsqueda se lo realizará en la columna `Descripción` de dataframe jobs_df.

In [11]:
# Value count of 'Descripción' column
skill = jobs_df['Descripción'].value_counts().reset_index()
skill

index  Descripción
0                                               Ninguno          791
1     Asignar rutas de entrega de mercadería y prior...          400
2               Disponibilidad de cambio de residencia:          161
3     Para importante industria en la ciudad de guay...          100
4     Buscamos persona para la comercialización de c...          100
...                                                 ...          ...
2178  Misión  Planificar, analizar y organizar la ob...            1
2179  Requisitos Título de tercer nivel en Contabili...            1
2180  Estamos en la busqueda de un programador front...            1
2181  Instructores docentes con certificaciones inte...            1
2182  Manejar camión para transportar cargas, materi...            1

[2183 rows x 2 columns]

In [12]:
possible_skills = ['.NET Framework',
                   'Accelerated Mobile Pages', 
                   'Active Server Pages',
                   'Adobe Creative Cloud',
                   'Adobe Illustrator',
                   'Adobe Photoshop',
                   'Adobe Premiere Pro',
                   'Amazon Web Services', 
                   'Angular',
                   'AngularJS',
                   'Ansible',
                   'Apache Hadoop',
                   'Apache Hive',
                   'Apache Kafka',
                   'Apache Maven',
                   'Apache Spark',
                   'App Store Optimization',
                   'Appium',
                   'Application programming interface', 
                   'Base de datos',
                   'Bootstrap',
                   'Business intelligence',
                   'CCNA',
                   'Cloud computing',
                   'Cloudera Platform',
                   'COBOL',
                   'CSS',
                   'Cucumber',
                   'Deep Learning',
                   'Design thinking',
                   'Docker',
                   'Email marketing',
                   'Extract, Transform and Load', 
                   'Firewall',
                   'Git',
                   'GNU/Linux',
                   'Google Ads',
                   'Google Analytics',
                   'Hacking',
                   'Hibernate',
                   'Hootsuite',
                   'HTML',
                   'IBM Db2 Family',
                   'Inteligencia Artificial',
                   'Intrusion prevention system',
                   'ITIL',
                   'Java',
                   'JavaScript',
                   'Jenkins',
                   'JIRA',
                   'jQuery',
                   'Kanban',
                   'Kotlin',
                   'Machine learning',
                   'Microservicios',
                   'Microsoft Azure',
                   'Microsoft Dynamics',
                   'Microsoft Office',
                   'Microsoft PowerPoint',
                   'Microsoft SQL Server',
                   'MicroStrategy',
                   'Model View Controller',
                   'MongoDB',
                   'MySQL',
                   'Node.js',
                   'Oracle RAC',
                   'PHP',
                   'PostgreSQL',
                   'PrestaShop',
                   'Project Management Professional', 
                   'PMP',
                   'Public Key Infrastructure', 
                   'PKI',
                   'Publicidad Digital',
                   'Python',
                   'Qlik',
                   'R programing language',
                   'React',
                   'Realidad virtual',
                   'Redes de comunicación',
                   'Redes sociales',
                   'SAP',
                   'Sass',
                   'Scala',
                   'Scripting language',
                   'Scrum',
                   'Search Engine Marketing',
                   'Search Engine Optimization',
                   'Security Information and Event Management', 
                   'Selenium',
                   'SEMrush',
                   'SOAP',
                   'Spring Framework',
                   'Swift',
                   'Tableau',
                   'TensorFlow',
                   'TypeScript',
                   'UiPath',
                   'Unity',
                   'Unix',
                   'User Experience',
                   'User Interface', 
                   'Visual Basic',
                   'VMware',
                   'WordPress',
                   'XML']

In [13]:
# The clean up process starts extracting the possible rules 
skills_group = skill['index'].str.extractall('({0})'.format('|'.join(possible_skills)))

# Create a column name
skills_group.columns = ['new_skill']

# Count the number of the possible rules and rename the dataframe
skills_group = pd.DataFrame(skills_group['new_skill'].value_counts()).reset_index()
skills_group.rename(columns={'index': 'skills', 'new_skill': 'count'}, inplace=True)

# Include the percentage
skills_group['percentage'] = skills_group['count'] / np.sum(skills_group['count'])
skills_group.head(20)

skills  count  percentage
0                         Java     14    0.145833
1                          PHP     10    0.104167
2                         HTML      9    0.093750
3                          SAP      9    0.093750
4                          CSS      7    0.072917
5             Microsoft Office      6    0.062500
6                    WordPress      3    0.031250
7           Microsoft Dynamics      3    0.031250
8                        Scrum      3    0.031250
9                      Angular      3    0.031250
10                         XML      2    0.020833
11                      Python      2    0.020833
12                        ITIL      2    0.020833
13                       React      2    0.020833
14              Redes sociales      2    0.020833
15                  Google Ads      2    0.020833
16                        CCNA      2    0.020833
17  Search Engine Optimization      1    0.010417
18             Microsoft Azure      1    0.010417
19                     Tableau      1    0.010417

## 5. Análisis de Centro de Capacitación

En la quinta parte del análisis, se considera el dataset de un centro de capacitación. Este archivo fue previamente procesado (cleaning and feature engineering) en R.

In [14]:
# read the csv file and parse dates
train_center = pd.read_csv('training_center_ec.csv', parse_dates=['Fecha_inicio', 'Fecha_fin', 'Fecha_registro',
                                                                 'Ultimo_login', 'Fecha_nacimiento'])
# eliminate 'Unnamed: 0' column
train_center.drop(['Unnamed: 0'], axis=1, inplace=True)

# change ID type to object
train_center['ID'] = train_center['ID'].astype('object')
train_center.head()

ID                     Curso Fecha_inicio  Fecha_fin       Estado  \
0  16807  Metodologias_Agiles_Lean          NaT        NaT  No iniciado   
1   6163  Metodologias_Agiles_Lean          NaT        NaT  No iniciado   
2  11583  Metodologias_Agiles_Lean   2019-12-04 2019-12-10     Aprobado   
3   7278  Metodologias_Agiles_Lean   2019-12-14 2019-12-15     Aprobado   
4  10890  Metodologias_Agiles_Lean          NaT        NaT  No iniciado   

   Puntuacion    Sexo Fecha_registro Ultimo_login Fecha_nacimiento   Ciudad  \
0         NaN  Hombre     2019-11-21   2019-11-26       1947-07-18  otavalo   
1         NaN  Hombre     2019-11-20   2019-11-20       1952-02-14    quito   
2        91.0  Hombre     2019-11-22   2020-02-06       1954-05-12    quito   
3        91.0   Mujer     2019-09-30   2019-12-15       1954-10-25    quito   
4         NaN   Mujer     2019-09-22   2019-09-25       1955-04-24    quito   

      Pais Desempleado   Nivel_estudios Tiempo_exp_laboral  \
0  Ecuador          SI  Máster/Posgrado      Más de 2 años   
1  Ecuador          NO     Licenciatura      Más de 2 años   
2  Ecuador          NO     Licenciatura                NaN   
3  Ecuador          NO     Licenciatura      Más de 2 años   
4  Ecuador          NO     Licenciatura      Más de 2 años   

  Tiempo_reg_serv_emp Tiempo_desemp   Edad  Tiempo_curso  Provincia  
0            No tiene      No tiene  72.87           NaN   Imbabura  
1                 NaN           NaN  68.29           NaN  Pichincha  
2                 NaN           NaN  66.05           7.0  Pichincha  
3                 NaN           NaN  65.59           2.0  Pichincha  
4       Más de 2 años           NaN  65.10           NaN  Pichincha

In [15]:
# check the dataframe `train_center` 
train_center.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8269 entries, 0 to 8268
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   ID                   8269 non-null   object        
 1   Curso                8269 non-null   object        
 2   Fecha_inicio         5726 non-null   datetime64[ns]
 3   Fecha_fin            4043 non-null   datetime64[ns]
 4   Estado               8269 non-null   object        
 5   Puntuacion           5657 non-null   float64       
 6   Sexo                 8269 non-null   object        
 7   Fecha_registro       8269 non-null   datetime64[ns]
 8   Ultimo_login         8269 non-null   datetime64[ns]
 9   Fecha_nacimiento     8269 non-null   datetime64[ns]
 10  Ciudad               8269 non-null   object        
 11  Pais                 8269 non-null   object        
 12  Desempleado          8269 non-null   object        
 13  Nivel_estudios       8269 non-nul

In [16]:
# Missing values in train_center dataframe
train_center.isna().sum()

ID                        0
Curso                     0
Fecha_inicio           2543
Fecha_fin              4226
Estado                    0
Puntuacion             2612
Sexo                      0
Fecha_registro            0
Ultimo_login              0
Fecha_nacimiento          0
Ciudad                    0
Pais                      0
Desempleado               0
Nivel_estudios            0
Tiempo_exp_laboral     3065
Tiempo_reg_serv_emp    4788
Tiempo_desemp          4532
Edad                      0
Tiempo_curso           4226
Provincia                 0
dtype: int64

In [17]:
# Let's check `Curso` column and count values of it
train_center['Curso'].value_counts()

Emprendimiento                   3070
Metodologias_Agiles_Lean         2033
Marketing_digital                 645
Wordpress                         327
Introduccion_Machine_Learning     317
Habilidades_socioemocionales      289
TICs_basico                       289
Diseno_web_HTML5_CSS              272
Growth_Hacking                    271
Habilidades_para_empleo           256
Introducción_programacion         206
Creacion_android_apps             187
Creacion_paginas_web               78
Analitica_web                      29
Name: Curso, dtype: int64

In [18]:
# What about the status of the trainings
train_center['Estado'].value_counts()

Aprobado       3728
No iniciado    2543
Iniciado       1683
Suspenso        315
Name: Estado, dtype: int64

In [19]:
# How many female and male are there?
train_center['Sexo'].value_counts()

Mujer     4289
Hombre    3980
Name: Sexo, dtype: int64

In [20]:
# How many attendees per Province?
train_center['Provincia'].value_counts()

Pichincha           2115
El Oro              1540
Guayas              1382
Imbabura             595
Manabi               342
Esmeraldas           294
Napo                 263
Azuay                259
Los Rios             220
Sucumbios            164
Chimborazo           147
Morona Santiago      127
Tungurahua           122
Loja                 122
Santo Domingo        116
Orellana             101
Santa Elena           87
Cotopaxi              78
Carchi                54
Bolivar               46
Cañar                 39
Zamora Chinchipe      27
Pastaza               23
Galapagos              6
Name: Provincia, dtype: int64

In [21]:
# Some statistics of numerical columns
train_center.describe()

Puntuacion         Edad  Tiempo_curso
count  5657.000000  8269.000000   4043.000000
mean     64.606152    30.962571     14.784071
std      37.694752    10.295029     16.567142
min       0.000000    17.510000      1.000000
25%      26.000000    23.420000      2.000000
50%      86.000000    27.580000      8.000000
75%      93.000000    36.500000     23.000000
max     100.000000    72.870000    157.000000

## 6. Código para graficar mapa en R

`mapdata <- get_data_from_map(download_map_data("countries/ec/ec-all"))
    data_ecuador <- mapdata %>% 
        select(PROVINCIA = `hc-a2`) %>% 
        mutate(X = c(195735, 17482, 5014, 15304, 7071, 16619, 18442, 31332,
                     19800, 14432, 2316, 5876, 4010, 12891, 20372, 52237,
                     15301, 2834, 226122, 2815, 23916, 6050, 3877, 	2811, 499))
    mapa1 <- hcmap("countries/ec/ec-all", data = data_ecuador, value = "X",
                   joinBy = c("hc-a2", "PROVINCIA"),
                   dataLabels = list(enabled = TRUE, format = '{point.name}'))`  